### ISSUE TITLE
User string interpolated into regex

### ISSUE THREAD

**Issue by mwest1066** [PrairieLearn/PrairieLearn#3448]
This causes a crash when the user string is itself an invalid regex.

```
node: const found = oldname.match(new RegExp(`^${basename} \\(copy ([0-9]+)\\)$`));
node: ^
node: SyntaxError: Invalid regular expression: /^CI for the population mean for paired observations) - The dioxin TCDD - Code \(copy ([0-9]+)\)$/: Unmatched ')'
node: at new RegExp (<anonymous>)
node: at /PrairieLearn/lib/editors.js:561:45
node: at Array.forEach (<anonymous>)
node: at getNumberLongName (/PrairieLearn/lib/editors.js:559:22)
node: at QuestionCopyEditor.getNamesForCopy (/PrairieLearn/lib/editors.js:575:32)
node: at /PrairieLearn/lib/editors.js:1248:34
node: at nextTask (/PrairieLearn/node_modules/async/dist/async.js:4576:27)
node: at next (/PrairieLearn/node_modules/async/dist/async.js:4584:13)
node: at /PrairieLearn/node_modules/async/dist/async.js:321:20
node: at /PrairieLearn/lib/editors.js:1243:21
node: at /PrairieLearn/lib/editors.js:517:13
node: at /PrairieLearn/lib/editors.js:509:21
node: at wrapper (/PrairieLearn/node_modules/async/dist/async.js:268:20)
node: at iteratorCallback (/PrairieLearn/node_modules/async/dist/async.js:495:17)
node: at /PrairieLearn/node_modules/async/dist/async.js:321:20
node: at /PrairieLearn/lib/editors.js:503:33
node: at /PrairieLearn/lib/editors.js:509:21
node: at wrapper (/PrairieLearn/node_modules/async/dist/async.js:268:20)
node: at iteratorCallback (/PrairieLearn/node_modules/async/dist/async.js:495:17)
node: at /PrairieLearn/node_modules/async/dist/async.js:321:20
node: at /PrairieLearn/lib/editors.js:498:29
node: at FSReqCallback.oncomplete (fs.js:156:23)
systemd: prairielearn.service: main process exited, code=exited, status=1/FAILURE
```

**Comment by andrewstec** [comment]
The `basename` has been escaped in https://github.com/PrairieLearn/PrairieLearn/pull/3385, but I need further understanding of this bug to test it.

**Comment by mwest1066** [comment]
> The `basename` has been escaped in #3385, but I need further understanding of where this bug occurs and how to recreate it to test it.

The fix in #3385 should take care of this.
